src/utilities/extendSchema.js

```diff
@@ -220,11 +213,21 @@ export function extendSchema(
         if (operationTypes[operation]) {
           throw new Error(`Must provide only one ${operation} type in schema.`);
         }
-        operationTypes[operation] = astBuilder.buildType(operationType.type);
+        const typeRef = operationType.type;
+        // Note: While this could make early assertions to get the correctly
+        // typed values, that would throw immediately while type system
+        // validation with validateSchema() will produce more actionable results.
+        operationTypes[operation] = (astBuilder.buildType(typeRef): any);
```

Comment:
Not sure how it worked previously since `astBuilder.buildType` returns `GraphQLNamedType` but `new GraphQLSchema` expects `query`/`mutation`/`subscription` to be `GraphQLObjectType`.
Looks like Flow bug 😞

Reply:
I'm guessing that the more declarative `operationTypes` definition has firmer types which flow can now report on where previously the error was hidden